src/app-logic/url-handling.js

```diff
@@ -80,27 +80,36 @@ function getDataSourceDirs(
   }
 }
 
-// "null | void" in the query objects are flags which map to true for null, and false
-// for void. False flags do not show up the URL.
-type BaseQuery = {|
-  v: number,
-  range: string, //
-  thread: string, // "3"
+// Base query that only applies to full profile view.
+type FullProfileSpecificBaseQuery = {|
   globalTrackOrder: string, // "3-2-0-1"
   hiddenGlobalTracks: string, // "0-1"
   hiddenLocalTracksByPid: string,
   localTrackOrderByPid: string,
-  file: string, // Path into a zip file.
-  transforms: string,
   timelineType: string,
   // The following values are legacy, and will be converted to track-based values. These
   // value can't be upgraded using the typical URL upgrading process, as the full profile
   // must be fetched to compute the tracks.
   threadOrder: string, // "3-2-0-1"
   hiddenThreads: string, // "0-1"
+|};
+
+// Base query that only applies to active tab profile view.
+type ActiveTabProfileSpecificBaseQuery = {||};
+
+// "null | void" in the query objects are flags which map to true for null, and false
+// for void. False flags do not show up the URL.
+type BaseQuery = {|
+  v: number,
+  range: string, //
+  thread: string, // "3"
+  file: string, // Path into a zip file.
+  transforms: string,
   profiles: string[],
   profileName: string,
   showTabOnly1: BrowsingContextID,
+  ...FullProfileSpecificBaseQuery,
+  ...ActiveTabProfileSpecificBaseQuery,
```

Comment:
So I changed the `Query` itself to use unions, but changing this to union made things A LOT worse to handle, so had to use it like this 😄